### PR TITLE
Loc rngfnd quality lua

### PR DESCRIPTION
**Pull request by ptrmu**

### CODE REVIEW

**Comment by ptrmu** [comment]
@clydemcqueen Here is another attempt at a lua interface that contains signal_quality_pct. If you get a chance, could you look this over and see if it makes sense. Thanks!

File: libraries/AP_RangeFinder/AP_RangeFinder_Lua.cpp [hunk 1]
```diff
@@ -28,25 +28,40 @@ AP_RangeFinder_Lua::AP_RangeFinder_Lua(RangeFinder::RangeFinder_State &_state, A
 }
 
 
-// Set the distance based on a Lua Script
-bool AP_RangeFinder_Lua::handle_script_msg(float dist_m)
+// Process range finder data from a lua driver. The three wriable items are copyied from
```
Comment (clydemcqueen) [inline libraries/AP_RangeFinder/AP_RangeFinder_Lua.cpp]: typos: writable, copied

File: libraries/AP_RangeFinder/AP_RangeFinder_Lua.cpp [hunk 2]
```diff
@@ -28,25 +28,40 @@ AP_RangeFinder_Lua::AP_RangeFinder_Lua(RangeFinder::RangeFinder_State &_state, A
...
+// state_arg to _state_pending. The update() method will copy this data into the backend
+// state structure. The get_state() method is called by a lau script to get the most recent
```
Comment (clydemcqueen) [inline libraries/AP_RangeFinder/AP_RangeFinder_Lua.cpp]: typo: lua

File: libraries/AP_RangeFinder/AP_RangeFinder_Lua.h [hunk 3]
```diff
@@ -20,15 +20,22 @@ class AP_RangeFinder_Lua : public AP_RangeFinder_Backend
     void update(void) override;
 
     // Get update from Lua script
-    bool handle_script_msg(float dist_m) override;
+    bool handle_script_msg(const RangeFinder::RangeFinder_State &state_arg) override;
```
Comment (clydemcqueen) [inline libraries/AP_RangeFinder/AP_RangeFinder_Lua.h:24]: Why not a smaller, writable-only struct? But perhaps this is the Lua binding design pattern, I don't know.

**Review by clydemcqueen — APPROVED** [review]
Very cool. I wonder about exposing RangeFinder::RangeFinder_State to the Lua interface, but this is used in other interfaces, so apparently it is the preferred pattern.

**Pull request closed without merging**